src/utils/references.ts

```diff
@@ -44,3 +45,47 @@ export function stringifyReferences (paths: string[], cwd: string): string {
 export function toReference (path: string, cwd: string): string {
   return `/// <reference path="${normalizeSlashes(isAbsolute(path) ? relative(cwd, path) : normalize(path))}" />`
 }
+
+export enum DependencyTarget {
+  Main,
+  Browser
+}
+
+export enum DependencyType {
+  Ambient,
+  External
+}
+
+export interface DependencyInfo {
+  target: DependencyTarget
+  type: DependencyType
+  name: string
+}
+
+export function parseReferencePath(path: string): DependencyInfo {
+  const referencePathRegexPattern =
```

Comment:
One final question (sorry). After spliting on `path.sep` and popping the file (index.d.ts), is it an error if the number of parts are != 3 (target, type, name) or is it possible for the name to contain `path.sep`? In other words, if number of parts > 3, join all parts after 2 to make the name.